crucible-cli/test-data/simulate/assumption-state.out.good

```diff
@@ -0,0 +1,143 @@
+==== Begin Simulation ====
+------------------------------
+empty
+-----
+Top-level goals:
+[]
+
+------------------------------
+base
+----
+Top-level goals:
+[]
+Assumptions:
+  in main test-data/simulate/assumption-state.cbl:10:5: assumption #0
+  casmp0@1:b
+
+------------------------------
+if
+--
+Top-level goals:
+[]
+Assumptions:
+  in main test-data/simulate/assumption-state.cbl:10:5: assumption #0
+  casmp0@1:b
+  Frame 4:
+    Assumptions:
+      [ The branch in main from test-data/simulate/assumption-state.cbl:18:5 to test-data/simulate/assumption-state.cbl:20:16
+      cb0@3:b
+      , in main test-data/simulate/assumption-state.cbl:21:5: assumption #1: if branch
+      casmp1@5:b ]
+
+------------------------------
+else
+----
+Top-level goals:
+[]
+Assumptions:
+  in main test-data/simulate/assumption-state.cbl:10:5: assumption #0
+  casmp0@1:b
+  Frame 8:
```

Comment:
It's a little counter-intuitive that these frame numbers are non-consecutive. The constructor for `AssumptionStack` takes a `NonceGenerator IO t` and `newOnlineBackend` initializes that to some nonce generator from the `ExprBuilder`... Might be better to just create a [`newIONonceGenerator`](https://hackage-content.haskell.org/package/parameterized-utils-2.1.10.0/docs/Data-Parameterized-Nonce.html#v:newIONonceGenerator) for each `AssumptionStack`? They're cheap (it is literally just a pointer to a 64-bit integer), and we only really expect there to be one `AssumptionStack` anyway.

Reply:
#1427